### PR TITLE
v2: remove deprecated targethost

### PR DESCRIPTION
**Pull request by hannahkm**
### What does this PR do?

Remove support for the `TargetHost` constant, which was deprecated in PR #1981 . Replace instances with `NetworkDestinationName` and `NetworkDestinationIP`.

A future PR will also remove support for the similar, also deprecated `PeerHostname` constant.

### Motivation

For v2 release, we should remove support for deprecated constants and functions, especially those that have been deprecated for many, subsequent minor versions.

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

**Comment by hannahkm** [comment]
TL;DR from offline conversation: we will keep support for `TargetHost` (and similar constants) and remove the deprecated label. Customers should still use `NetworkDestinationName` and `NetworkDestinationIP` when possible.
